Mithril update 2023-11-29 (#298)
input-output-hk/cardano-updates: blog/2023-11-29-mithril.md

---
title: Mithril Team Update
slug: 2023-11-29-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> This week, the Mithril team released the first version of the [Mithril client library](https://crates.io/crates/mithril-client) which allows developers to implement the core features of Mithril in their own applications. The team also released a new distribution [`2347.0`](https://github.com/input-output-hk/mithril/releases/tag/2347.0) which includes support for slim certificates production by the aggregator, as well as some bug fixes and performance improvements.

Reviewer: ```suggestion
This week, the Mithril team released the initial version of the [Mithril client library](https://crates.io/crates/mithril-client), enabling developers to integrate core Mithril features into their applications. They also released a new distribution, [`2347.0`](https://github.com/input-output-hk/mithril/releases/tag/2347.0), which includes support for slim certificate production by the aggregator, along with bug fixes and performance improvements.
```

> The team has also rolled-out of a first version of the Cardano/Mithril node communications enhancements implemented by TxPipe as part of a Catalyst project. They made progress in decentralizing the Mithril networks with the testing of the Mithril P2P relay on a test Mithril network, and the elaboration of a threat modeling and risk analysis for peer-to-peer (P2P) networking. They also started working on adapting the Mithril client library so that it supports WASM compilation. Additionally, they have made progress in optimizing the performances of the aggregator. 

Reviewer: ```suggestion
The team also introduced the initial version of Cardano/Mithril node communication enhancements,  implemented by TxPipe as part of the Catalyst project. They made progress in decentralizing Mithril networks, including testing the Mithril peer-to-peer (P2P) relay on a test network and conducting threat modeling and risk analysis for  P2P networking. Additionally, they started working on adapting the Mithril client library for WASM compilation and made progress in optimizing aggregator performance. 
```

> Finally, they fixed the verbosity level of the logs which was not accurate in most of the nodes, and they worked on troubleshooting some users.

Reviewer: ```suggestion
Finally, they fixed inaccuracies in the verbosity level of logs across most nodes and worked on troubleshooting for some users.
```
